ts/private/ts_project.bzl

```diff
@@ -232,16 +231,26 @@ See https://github.com/aspect-build/rules_ts/issues/361 for more details.
         typecheck_output = ctx.actions.declare_file(ctx.attr.name + ".typecheck")
         typecheck_outs.append(typecheck_output)
 
-        typecheck_arguments = ctx.actions.args()
-        if supports_workers:
-            typecheck_arguments.add("--bazelValidationFile", typecheck_output.short_path)
+        args = ctx.actions.args()
+        args.add_all(common_args)
 
-        typecheck_arguments.add("--noEmit")
+        args.add("--noEmit")
+
+        env = {
+            "BAZEL_BINDIR": ctx.bin_dir.path,
+        }
+
+        if supports_workers:
+            args.use_param_file("@%s", use_always = True)
+            args.set_param_file_format("multiline")
+            args.add("--bazelValidationFile", typecheck_output.short_path)
+        else:
+            env["JS_BINARY__STDOUT_OUTPUT_FILE"] = typecheck_output.path
```

Comment:
It took me several hours to finally narrow down to this as the root cause of a bug where running a command with workers enabled was just hanging. The persistent worker protocol uses stdout to communicate with the worker process, but setting this env variable makes stdout of the worker process get captured into a temp file by the `js_binary` wrapper script:
https://github.com/aspect-build/rules_js/blob/f583319639a7875b891ab35701a3d0e630d4f8f3/js/private/js_binary.sh.tpl#L21-L23
So the Bazel process was never receiving the message that the worker was done with the job.
The purpose of capturing stdout was to create the marker file, so when running in worker mode we instead rely on the `--bazelValidationFile` flag (see handling in ts_project_worker.js).

Interesting to note is that the logic was originally structured similarly in the original validation action PR:
https://github.com/aspect-build/rules_ts/pull/574/files#r1744177502

Reply:
Interesting!

Can we maybe add a comment here explaining that a little bit? Even just essentially "The persistent worker protocol uses stdout to communicate with the worker process so stdout can only be redirected in non-worker mode"